sui_programmability/examples/nft_standard/README.md

```diff
@@ -0,0 +1,114 @@
+# NFT Standards, Philosophy, and Requirements
+
+# Basics
+
+Sui’s object-centric data model gives Sui objects many of the features of a basic ERC721-style NFT: an owner, the ability to be transferred, a globally unique identifier.
+
+### Individual NFT’s
+
+An NFT is a Sui object (i.e., a struct value whose declared type has the `key` ability, which in turn means the struct value has a field named `id` holding a globally unique ID). All NFT’s are Sui objects, but not every Sui object is an NFT.
+
+### Collections
+
+An NFT collection is a Move struct type with the `key` ability. Every NFT collection has a distinct Move struct type, but not every Move struct type corresponds to an NFT collection.
+
+Collection metadata is a singleton object of type `sui::collection::Collection<T>` created via the [module initializer](https://examples.sui.io/basics/init-function.html) of the module that defines the NFT type `T`. Initializing a collection gives the creator a `MintCap<T>` granting permission to mint NFT's of the given type, and a `RoyaltyCap<T>` granting permission to mint `RoyaltyReceipt<T>`'s (which will be required upon sale--more on this below).
```

Comment:
- We have given much thought to this. Our current perspective is that if we force all collections to have a Limited Supply via `MintCap<T>` we could risk leaving out some domain-specific use cases.

- One use case could be a gaming item that can be minted on demand by the Gaming Creators. If the game wants unlimited supply and does not care about keeping track of the amount of objects minted in the collection object, we can take the most advantage out of parallelisation because we don’t require each mint transaction to mutate the supply in the `Collection`/`MintCap` object.

Reply:
- Definitely agree with the desire to allow unlimited supply as an option.
- The convention I had in mind for unlimited supply is to set the collection supply to `U64_MAX`. No one can feasibly mint that many NFT's.
- The prototype code does not require touching the `Collection` object to mint, for exactly the reasons you mention (want to max out parallelization). I don't think the requirement to touch a `MintCap` to mint is a barrier to parallelization--if you don't want a single `MintCap` to be a point of contention, you can always split it into as many sub `MintCap`'s as desired.